src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1951,8 +1951,14 @@ def client_side_proxy_wrapper(cmd,
     tenant_id = profile.get_subscription()['tenantId']
 
     client_proxy_port = consts.CLIENT_PROXY_PORT
+
+    # Check if the internal port is already open. If so and the user has specified a port,
+    # try using the port before the user specified port instead.
+    if clientproxyutils.check_if_port_is_open(client_proxy_port) and api_server_port != consts.API_SERVER_PORT:
+        client_proxy_port = int(api_server_port) - 1
```

Comment:
Just a thought, could following be a better user experience? And probably won't need next `if` condition as well.
```
# Check if the internal port is already open. If so, try to find unused port
while clientproxyutils.check_if_port_is_open(client_proxy_port) or client_proxy_port == api_server_port:
    client_proxy_port = client_proxy_port - 1
```